### PR TITLE
Use xterm.js for log outputs

### LINKED ISSUE #718: The spaces in task output should be kept.
### Component

web-ui

### Describe the bug

The spaces in task output should be kept.

image

### System Info

```shell
"source": "https://github.com/woodpecker-ci/woodpecker",
"version": "next-9feb6be9"
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy).

### LINKED ISSUE #768: Browser tab crashes when long output printed and we attempt to monitor it in real time
### Component

web-ui

### Describe the bug

On output with 5500 lines, clicking pipeline which is producing these lines in realtime browser gets frozen with message "Woodpecker is not reposinding [WAIT]". After build is done and we reopen woodpecker in different tab we able to see an output.

### System Info

```shell
0.15.32
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy).

### LINKED ISSUE #776: cut / tail very long logs
We currently send all log data at once to the client. In the case a job generated a lot of logs we should for example think about just sending the last 500 lines of logs otherwise the browser tab will OOM when dealing with that much data. Chunking will probably not help in this case as well as the browser would have to load that data in memory. Maybe we can set a limit by default for the api endpoint and allow to download / receive the whole data (#566) when a parameter is applied.

### PR DESCRIPTION
**Pull request by anbraten**
closes #718
closes #768 
closes #776
solves parts of #721 

# Fixes
- flickering when adding new log-lines while step is running

### CODE REVIEW

**Comment by 6543** [comment]
downside: log line numbers are gone

**Comment by anbraten** [comment]
> downside: log line numbers are gone

and the time on the right. Maybe we could create a xterm.js plugin to get this functionality back in the long term

**Comment by 6543** [comment]
> ... we could create a xterm.js plugin ...

I think we should do this first :)

**Comment by x80486** [comment]
I was [sniffing](xtermjs/xterm.js/issues/2016) on the `Xterm.js` issue tracker and sort of make sense that this data should be handled outside of the library. I guess a couple of functions will do it as well.

There is an example on the aforementioned issue.

**Comment by anbraten** [comment]
I added the loglines similar to the suggestion from the xterm issue. Not super happy with their styling, but I think its fine like that for now.

Screenshot from 2022-06-03 18-17-59

File: web/src/assets/locales/en.json [hunk 1]
```diff
@@ -171,6 +171,8 @@
       "no_pipelines": "No pipelines have been started yet.",
       "step_not_started": "This step hasn't started yet.",
       "pipelines_for": "Pipelines for branch \"{0}\"",
+      "exit_code": "exit code {exitCode}",
```
[the remark this answers is not in the document]
Comment (anbraten) [reply web/src/assets/locales/en.json:174]: You are right, we should stay consistent here. As the object style is easier to understand for 2 or more arguments I would prefer to use it for single arguments as well to be even more consistent.  I changed 168 and 173 to the object style as well.

**Comment by 6543** [comment]
I'm going to test it today

**Comment by 6543** [comment]
https://cloud.obermui.de/s/RZg582XmKG4SN48

the background colloring has some strang relicts

**Comment by 6543** [comment]
and switching lgitht/dark-mode do result in unpredictable behaviour :/

**Comment by anbraten** [comment]
Seems to work for me. Maybe browser related I would do a follow-up in that case if that is fine for you.

https://user-images.githubusercontent.com/6918444/173768959-afd42db2-027f-4fe2-bf76-d365f1cedf3b.mp4

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
great work :rocket: i would say - finally :)

**Pull request merged**